### PR TITLE
Enable Client Deduplication ping anonymization

### PR DESCRIPTION
**Pull request by fbertsch**
The outcome here is the new tables should reflect almost exactly what our ping structure is. One difference is the addition of submission_date for ease of querying / partitioning.
- Hash the Ad IDs with a secret key
- Replace stable views with views on derived tables
- Add view over all client deduplication pings

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/client_deduplication_v1/query.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
+-- Query for org_mozilla_firefox_derived.client_deduplication_v1
+WITH hmac_key AS (
+  SELECT
+    AEAD.DECRYPT_BYTES(
+      (SELECT keyset FROM `moz-fx-dataops-secrets.airflow_query_keys.org_mozilla_firefox`),
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/client_deduplication_v1/query.sql:5]:
```suggestion
      (SELECT keyset FROM `moz-fx-dataops-secrets.airflow_query_keys.ad_id_prod`),
```

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/client_deduplication_v1/query.sql [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      ciphertext,
+      CAST(key_id AS BYTES)
+    ) AS value
+  FROM
+    `moz-fx-data-shared-prod.org_mozilla_firefox.encrypted_keys_v1`
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/client_deduplication_v1/query.sql:10]:
```suggestion
    `moz-fx-data-shared-prod.org_mozilla_firefox_derived.encrypted_keys_v1`
```

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/client_deduplication_v1/query.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  WHERE
+    key_id = 'ad_id'
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/client_deduplication_v1/query.sql:12]:
```suggestion
    key_id = 'ad_id_prod'
```

**Review by whd — APPROVED** [review]
The suggestions needs to be propagated to the other queries but otherwise LGTM. 

CI seems busted though.

File: sql/moz-fx-data-shared-prod/org_mozilla_firefox_beta_derived/client_deduplication_v1/query.sql [hunk 4]
```diff
@@ -0,0 +1,38 @@
+-- Query for org_mozilla_firefox_beta_derived.client_deduplication_v1
+WITH hmac_key AS (
+  SELECT
+    AEAD.DECRYPT_BYTES(
+      (SELECT keyset FROM `moz-fx-dataops-secrets.airflow_query_keys.org_mozilla_firefox`),
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/org_mozilla_firefox_beta_derived/client_deduplication_v1/query.sql:5]:
```suggestion
      (SELECT keyset FROM `moz-fx-dataops-secrets.airflow_query_keys.ad_id_prod`),
```

File: sql/moz-fx-data-shared-prod/org_mozilla_firefox_beta_derived/client_deduplication_v1/query.sql [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      ciphertext,
+      CAST(key_id AS BYTES)
+    ) AS value
+  FROM
+    `moz-fx-data-shared-prod.org_mozilla_firefox.encrypted_keys_v1`
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/org_mozilla_firefox_beta_derived/client_deduplication_v1/query.sql:10]:
```suggestion
    `moz-fx-data-shared-prod.org_mozilla_firefox_derived.encrypted_keys_v1`
```

File: sql/moz-fx-data-shared-prod/org_mozilla_firefox_beta_derived/client_deduplication_v1/query.sql [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  WHERE
+    key_id = 'ad_id'
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/org_mozilla_firefox_beta_derived/client_deduplication_v1/query.sql:12]:
```suggestion
    key_id = 'ad_id_prod'
```

File: sql/moz-fx-data-shared-prod/org_mozilla_firefox_derived/client_deduplication_v1/query.sql [hunk 7]
```diff
@@ -0,0 +1,38 @@
+-- Query for org_mozilla_firefox_derived.client_deduplication_v1
+WITH hmac_key AS (
+  SELECT
+    AEAD.DECRYPT_BYTES(
+      (SELECT keyset FROM `moz-fx-dataops-secrets.airflow_query_keys.org_mozilla_firefox`),
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/org_mozilla_firefox_derived/client_deduplication_v1/query.sql:5]:
```suggestion
      (SELECT keyset FROM `moz-fx-dataops-secrets.airflow_query_keys.ad_id_prod`),
```

File: sql/moz-fx-data-shared-prod/org_mozilla_firefox_derived/client_deduplication_v1/query.sql [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      ciphertext,
+      CAST(key_id AS BYTES)
+    ) AS value
+  FROM
+    `moz-fx-data-shared-prod.org_mozilla_firefox.encrypted_keys_v1`
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/org_mozilla_firefox_derived/client_deduplication_v1/query.sql:10]:
```suggestion
    `moz-fx-data-shared-prod.org_mozilla_firefox_derived.encrypted_keys_v1`
```

File: sql/moz-fx-data-shared-prod/org_mozilla_firefox_derived/client_deduplication_v1/query.sql [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  WHERE
+    key_id = 'ad_id'
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/org_mozilla_firefox_derived/client_deduplication_v1/query.sql:12]:
```suggestion
    key_id = 'ad_id_prod'
```

**Pull request merged**